op-guide/monitor.md

```diff
@@ -0,0 +1,157 @@
+        "store_id": 1,               // 节点的 ID
+        "capacity": 1968874332160,   // 存储总容量
+        "available": 1264847716352,  // 存储剩余容量
+        "region_count": 1,           // 该节点上存放的  Region 数量
+        "sending_snap_count": 0,
+        "receiving_snap_count": 0,
+        "start_ts": "2016-10-24T19:54:00.110728339+08:00",            // 启动时间
+        "last_heartbeat_ts": "2016-10-25T10:52:54.973669928+08:00",   // 最后一次心跳时间
+        "total_region_count": 1,           // 总 Region 数量
+        "leader_region_count": 1,        // Leader Region 数量
+        "uptime": "14h58m54.862941589s"
+      },
+      "scores": [
+        100,
+        35
+      ]
+    }
+  ]
+}
+```
+
+## Metrics 监控
+这部分主要对整个集群的状态、性能做监控，通过 prometheus+grafana 展现 metrics 数据，在下面一节会介绍如何搭建监控系统。
+
+### TiDB Server
++ query 处理时间，可以看到延迟和吞吐
++ ddl 过程监控
++ TiKV client 相关的监控
++ PD client 相关的监控
+
+### PD Server
++ 命令执行的总次数
++ 某个命令执行失败的总次数
++ 某个命令执行成功的耗时统计
++ 某个命令执行失败的耗时统计
++ 某个命令执行完成并返回结果的耗时统计
+
+### TiKV Server
+
++ GC 监控
++ 执行 KV 命令的总次数
++ Scheduler 执行命令的耗时统计
++ Raft propose 命令的总次数
++ Raft 执行命令的耗时统计
++ Raft 执行命令失败的总次数
++ Raft 处理 ready 状态的总次数
+
+## 使用 Prometheus+Grafana
+### 部署架构
+整个架构如下图所示，在 TiDB/PD/TiKV 三个组件的启动参数中添加 Prometheus Pushgateway 地址
+![architecture](./monitor.png)
+
+### 搭建监控系统
+Prometheus Push Gateway
+参考：https://github.com/prometheus/pushgateway
+
+Prometheus Server
+参考： https://github.com/prometheus/prometheus#install
+
+Grafana
+参考：http://docs.grafana.org
+
+
+### 配置
+#### TiDB/PD/TiKV 配置
++ TiDB
+设置 --metrics-addr 和 --metrics-interval 两个参数，其中 metrics-addr 设为 push gateway 的地址，metrics-interval 为 push 的频率，单位为秒，默认值为15
++ PD
+修改配置文件，填写 push gateway 的地址和推送频率
+```
+[metric]
+# prometheus client push interval, set "0s" to disable prometheus.
+interval = "15s"
+# prometheus pushgateway address, leaves it empty will disable prometheus.
+address = "host:port"
+```
+
++ TiKV
+修改配置文件，填写 push gateway 的地址和推送频率，job 字段一般设为“tikv”
+```
+[metric]
+# the Prometheus client push interval. Setting the value to 0s stops Prometheus client from pushing.
+interval = "15s"
+# the Prometheus pushgateway address. Leaving it empty stops Prometheus client from pushing.
+address = "host:port"
+# the Prometheus client push job name. Note: A node id will automatically append, e.g., "tikv_1".
+job = "tikv"
+```
+
+#### PushServer 配置
+一般无需特殊配置，使用默认端口 9091 即可
+ 
++ Prometheus 配置
+在配置文件中添加 pushgateway 地址：
+```
+scrape_configs:
+  # The job name is added as a label `job=<job_name>` to any timeseries scraped from this config.
+  - job_name: 'TiDB'
+
+
+    # Override the global default and scrape targets from this job every 5 seconds.
+    scrape_interval: 5s
+
+
+    static_configs:
+      - targets: ['host:port'] # 这里填写 pushgateway 地址
+        labels:
+                group: 'production'
+```
+#### Grafana 配置
+```
+进入 grafana Web 界面 ，默认3000端口
+    添加 datasource:
+  图标 Logo -> Data Sources -> Add data source
+            Type 选择 Promethus，url 填 promethus address，Access 选择 direct
+             将 Default 勾选上
```

Comment:
I find a way  select data source when import json.we do not need to set default.